ci-operator/step-registry/ibmcloud/deprovision/resourcegroup/ibmcloud-deprovision-resourcegroup-commands.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output   
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..." 
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function run_command_with_retries() {
+  cmd="$1"
+  retries="$2"
+  interval="$3"
+
+  if [ X"$retries" == X"" ]; then
+      retries=20
+  fi
+
+  if [ X"$interval" == X"" ]; then
+      interval=30 
+  fi
+
+  set +o errexit
+
+  output=$(eval "$cmd"); ret=$?
+  try=1
+  while [ X"$ret" != X"0" ] && [ $try -lt $retries ]; do
+      sleep $interval
+      output=$(eval "$cmd"); ret=$?
+      try=$(expr $try + 1)
+  done
+
+  set -o errexit
+
+  if [ X"$try" == X"$retries" ]; then
+      return 2
+  fi
+  echo "$output"
+  return 0
+}
+
+ibmcloud_login
+
+rg_files="${SHARED_DIR}/ibmcloud_resource_group"
+for rg_file in ${rg_files}; do
```

Comment:
not just one file named "${SHARED_DIR}/ibmcloud_resource_group", saved the resource group name?

Reply:
in the future, we can easily extend the file list, so use for loop here.

Reply:
now have not a step which creates file ${SHARED_DIR}/ibmcloud_cluster_resource_group? 
how about hiding the code which may extend in the future? in order to be easy to understand?

Reply:
> now have not a step which creates file ${SHARED_DIR}/ibmcloud_cluster_resource_group?

No, I never saw that.

> how about hiding the code which may extend in the future? in order to be easy to understand?

I already added it when resolving https://github.com/openshift/release/pull/49541#discussion_r1514751928.